### PR TITLE
Add support for TET14 geometric elements

### PR DESCRIPTION
**Pull request by nmnobre**
## Reason
This is the second stepping stone on the road to Raviart-Thomas support.

## Design
This is much simpler than the TRI7 PR mostly due to the consolidation work we've been doing. Unfortunately, half the changes are still for adding a test file which is mostly redundant, but I see no clean way around it (#25829)... :/

Importantly, there's a couple of tests for adaptive mesh refinement which will definitely fail.
This is good, or at least expected, and will serve as a downstream test for libMesh/libmesh#3649.
Once the libMesh submodule is brought up to date the tests should (will 🤞) pass.

## Impact
By itself, this PR should enable P2 with bubble (aka P2+) simulations on 3d tetrahedral meshes.

### CODE REVIEW

**Review by roystgnr — APPROVED** [review]
I don't know enough about the ray tracing module, but unless @loganharbour sees an issue there, the rest of this looks excellent to me.

File: modules/xfem/src/efa/EFAElement3D.C [hunk 1]
```diff
@@ -33,12 +33,14 @@ EFAElement3D::EFAElement3D(unsigned int eid, unsigned int n_nodes, unsigned int
   if (_num_faces == 4)
   {
     _num_vertices = 4;
-    if (_num_nodes == 10)
+    if (_num_nodes == 14)
+      _num_interior_face_nodes = 4;
+    else if (_num_nodes == 10)
       _num_interior_face_nodes = 3;
```
Comment (GiudGiud) [inline modules/xfem/src/efa/EFAElement3D.C:39]: wait is that true

Comment (GiudGiud) [reply modules/xfem/src/efa/EFAElement3D.C:39]: tet10 is not 4 vertices + 6 edge vertices?

Comment (nmnobre) [reply modules/xfem/src/efa/EFAElement3D.C:39]: > tet10 is not 4 vertices + 6 mid-edge nodes?

Indeed. And then you add another node for the face to form a TET14.

Comment (GiudGiud) [reply modules/xfem/src/efa/EFAElement3D.C:39]: so should be 0 not 3? (this is not from your PR I know)

**Comment by GiudGiud** [comment]
No priority system that I know of in CIVET. Others might be just lucky

**Comment by nmnobre** [comment]
> No priority system that I know of in CIVET. Others might be just lucky

For instance, the Conda build tests which should take 3 or 4 mins are taking more than 40 mins?!

File: modules/xfem/src/efa/EFAElement3D.C (hunk 1 above)
Comment (GiudGiud) [reply modules/xfem/src/efa/EFAElement3D.C:39]: oh so it s on a per-face basis?

Comment (nmnobre) [reply modules/xfem/src/efa/EFAElement3D.C:39]: Indeed.

Comment (GiudGiud) [reply modules/xfem/src/efa/EFAElement3D.C:39]: all good then

**Comment by nmnobre** [comment]
> I don't know enough about the ray tracing module, but unless @loganharbour sees an issue there, the rest of this looks excellent to me.

Perfect! 🤗
We now need to update the libMesh submodule.
I see you usually put together a (not so short) changelog in the commit message, are you happy to do that, or do you want me to attempt an initial draft?

**Comment by GiudGiud** [comment]
It's usually worth splitting the libmesh update in a separate PR to avoid having to deal with multiple necessary changes in an unrelated PR.
@roystgnr or @lindsayad you planning a libmesh update?

For the failing test you ll have to check if it s the rebase on next that s the cause or your own code

**Comment by nmnobre** [comment]
> It's usually worth splitting the libmesh update in a separate PR to avoid having to deal with multiple necessary changes in an unrelated PR.

Absolutely! If I were to give it a shot, I wouldn't do it as part of this PR. :)

> For the failing test you ll have to check if it s the rebase on next that s the cause or your own code

The tests won't pass until we use a newer version of libMesh, they depend on a fairly recent bug fix, https://github.com/libMesh/libmesh/pull/3649.

**Comment by roystgnr** [comment]
libMesh update this week.  We've got two outstanding PRs (and one outstanding investigation I'm *hoping* to put into a PR) that I'd love to slip in if we can, but if not then IMHO the VTK 9.3 compatibility fix was the last thing that really *needed* to get in to the update.

**Review by lindsayad — APPROVED** [review]

**Pull request merged**